### PR TITLE
Add a @_borrowed attribute to force the use of an opaque read accessor

### PR DESCRIPTION
**Pull request by rjmccall**
This is strawman syntax pending a proposal; I pitched some ideas here:
  https://forums.swift.org/t/value-ownership-when-reading-from-a-storage-declaration/15076

### CODE REVIEW

File: lib/Sema/TypeCheckDecl.cpp [hunk 1]
```diff
@@ -2341,8 +2341,36 @@ static bool computeIsSetterMutating(TypeChecker &TC,
...
+static bool shouldUseOpaqueReadAccessor(TypeChecker &TC,
+                                        AbstractStorageDecl *storage) {
+  // If the attribute isn't present, do nothing.
+  if (!storage->getAttrs().hasAttribute<BorrowedAttr>())
+    return false;
+
+  // The start of this logic follows
+  // AbstractStorageDecl::requiresOpaqueModifyCoroutine().
+
+  // These criteria are basically required for correctness.
```
Comment (slavapestov) [inline lib/Sema/TypeCheckDecl.cpp]: Should they not be enforced via diagnostics instead then?

Comment (rjmccall) [reply lib/Sema/TypeCheckDecl.cpp]: Look at you having high expectations. :)

**Comment by rjmccall** [comment]
@swift-ci Please smoke test

**Comment by rjmccall** [comment]
@swift-ci Please smoke test Linux

**Comment by rjmccall** [comment]
I don't really believe this patch broke random LLDB tests, but I don't know what else to do here.

**Comment by rjmccall** [comment]
@swift-ci Please smoke test.

**Pull request merged**